src/D2L.CodeStyle.Analyzers/Helpers/ExemptSymbolsBuilder.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+
+namespace D2L.CodeStyle.Analyzers.Helpers;
+
+internal sealed class ExemptSymbolsBuilder {
+
+	private readonly Compilation m_compilation;
+	private readonly AnalyzerOptions m_analyzerOptions;
+	private readonly CancellationToken m_cancellationToken;
+
+	private readonly ImmutableHashSet<ISymbol>.Builder m_exemptions = ImmutableHashSet.CreateBuilder<ISymbol>( SymbolEqualityComparer.Default );
+
+	[System.Diagnostics.CodeAnalysis.SuppressMessage(
+		"MicrosoftCodeAnalysisPerformance",
+		"RS1012:Start action has no registered actions",
+		Justification = "Not an analyzer"
+	)]
+	public ExemptSymbolsBuilder(
+		CompilationStartAnalysisContext context
+	) {
+		m_compilation = context.Compilation;
+		m_analyzerOptions = context.Options;
+		m_cancellationToken = context.CancellationToken;
+	}
+
+	public ImmutableHashSet<ISymbol> Build() => m_exemptions.ToImmutable();

```

Comment:
`IReadOnlySet<T>`/`HashSet<T>` is better performance

Reply:
(`ImmutableHashSet<T>` uses an implementation that makes functional edits cheaper, at the cost of making lookups slower.)